docs/rfcs/1023-async-enrollment.md

```diff
@@ -0,0 +1,1093 @@
+<!-- Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 2016-present Scille SAS -->
+
+# Asynchronous enrollment
+
+## 1 - Goals
+
+User enrollment into an organization is divided into two steps:
+
+1. Checking the identity of both greeter and claimer.
+2. Actually registering the claimer as a member of the organization.
+
+Currently those two steps are done in a synchronous way from the users' point of view:
+
+- Both claimer and greeter must be online at the same time.
+- Claimer and greeter have to manually exchange some information (i.e. SAS code exchange).
+
+Those two requirements are needed to fulfill step 1 when both claimer and greeter know
+nothing of their respective peer.
+
+However this is not always the case: a 3rd party service may be used to fulfill step 1,
+which in turn open the door to an all-automatic enrollment process.
+
+## 2 - Overview
+
+### 2.1 - Submitter/accepter vs claimer/greeter
+
+In regular synchronous enrollment, the two peers are designed as claimer
+(the one that wants to be enrolled) and greeter (the one that does the enrollment).
+
+However this term is changed to submitter & accepter in the asynchronous enrollment.
+This is notably to reflect two fundamental differences:
+
+- All administrator in the organization can deal with any given asynchronous enrollment (while
+  there is only a single greeter per synchronous enrollment).
+- An asynchronous enrollment is created by the submitter (while a synchronous one is created
+  by the greeter).
```

Comment:
```suggestion
- When the submitter creates an enrollment it is asynchronous, whereas when an accepter creates the enrollment, it is synchronous.
```
It kinda sounded like the people involved could choose what kind of enrollment they wanted, but maybe it's just me. (Also greeter -> accepter)

Reply:
> It kinda sounded like the people involved could choose what kind of enrollment they wanted

That's the point here: this paragraph show the different between the synchronous enrollment (i.e. the one we currently have with SASCode) and an asynchronous one.

> Also greeter -> accepter

Here we are precisely talking about the peer that creates the new user in a synchronous enrollment, so it is the greeter ^^


Anyway, I've rephrased this part to make it more clear :+1: